src/amplitude-client.js

```diff
@@ -75,6 +75,12 @@ AmplitudeClient.prototype.init = function init(apiKey, opt_userId, opt_config, o
     return;
   }
 
+  var hasExistingCookie = !!this.cookieStorage.get(this.options.cookieName + this._storageSuffix);
+  if (opt_config && opt_config.deferInitialization && !hasExistingCookie) {
```

Comment:
ok so let's say a user accepts and allows tracking and revisits the page the next day. There would already be an existing cookie and we would skip _deferInitialization. However, wouldn't `this._shouldDeferCall()` still return true?

Reply:
No `this._shouldDeferCall()` would return `false` (for the most part, there's a short period of time in the `init` function where `this._pendingReadStorage` returns true while getting started).

It depends on how they revisit the page. If the page never refreshed and they have the same instance of Amplitude, then the new events will not be sent --- I've added a test for this

If it's a new instance of amplitude (e.g refresh page, navigate to another page), then `hasExistingCookie` will be true and `_initializationDeferred ` will never be set to true, thus `this._shouldDeferCall()` will be false as long as `this._pendingReadStorage` is also false (most of the time)

Reply:
actually, after fixing my tests, i realized that you were right! gratzi!

Reply:
After adding tests, and further inspecting, i realize you're right. Because `_initializationDeferred` is never set to false AND `this._storageSuffix` isn't defined until the try block below, it's always assuming `_initializationDeferred` is true.

I've added a test to reflect it and additional assert statements for the existing tests